### ISSUE TITLE
Slow video loading on Matrix

### ISSUE THREAD

**Issue by yekul** [croneter/PlexKodiConnect#1531]
Had issues with firestick in recent months, where video loads are extremely slow to happen. Click a video, nothing happens for what feels like eternity. Been wasting so much time on it and finally loaded back up Leia, along with the older beta repo for PKC and videos now loading instantly again. 

Is this a known issue of some sort? I will note the same issue occurs with the latest version of Composite installed as well (never used it though tbh, just wanted to test something else). 

Has been driving me absolutely insane but can't seem to find other people whinging about it so feels like a very 'me' issue right now. Attached my kodi log, with the hopes that's what is needed to help diagnose? 

I've performed a reset, multiple data wipes, multiple devices. Working off Windows now to make it easier to access everything. 

[kodi.txt](https://github.com/croneter/PlexKodiConnect/files/6822497/kodi.txt)

**Comment by croneter** [comment]
Indeed, it takes forever for Kodi to start playback once PKC told it what to play: 2.5 minutes!
```
20:31:47.255 T:4800     INFO <general>: PLEX.playback_decision: The playurl for DirectPlay is: https://...
...
2021-07-15 20:34:08.958 T:6276     INFO <general>: Creating Demuxer
2021-07-15 20:34:09.076 T:6276     INFO <general>: Opening stream: 0 source: 256
```
This is not normal but I'm afraid once PKC told Kodi what to play, it's entirely up to Kodi. Can you please share a debug log?

**Comment by yekul** [comment]
I will note it isn't that bad on the firestick, usually around ~20s or so (which is still terrible compared to instantly, but on the computer which I set up purely to test this it is pretty terrible). 

Debug log attached. Appreciate any and all insight you can provide, as it feels like a local setup issue but I really haven't changed many default settings so not too sure what it could be. As I said, Leia works without issue, loading as quick as the Plex app (which also works instantly). 

[kodidebug.txt](https://github.com/croneter/PlexKodiConnect/files/6826512/kodidebug.txt)

**Comment by yekul** [comment]
Probably not very useful, but just downgraded the firestick to Leia (18.9 iirc) and Leia Beta PKC install and also working as expected (~1s delay launching things). So something is definitely unhappy having made the move to Matrix, but just not quite sure what.

**Comment by croneter** [comment]
That's not the entire debug log, @yekul . Kindly share an entire debug log

**Comment by croneter** [comment]
The issue seems to be Kodi (NOT PKC) scanning for subtitles:
```
08:05:40.114 T:6784    DEBUG <general>: CUtil::ScanForExternalSubtitles: Searching for subtitles...
08:05:40.115 T:6784    DEBUG <general>: CurlFile::Open(0x1cafbef0) https://192-168-1-69.04149d80325546f2ae651c03a3597f10.plex.direct:32400/library/parts/21957/1573239338/
08:05:40.115 T:6784    DEBUG <general>: XCURL::DllLibCurlGlobal::easy_acquire - Created session to https://192-168-1-69.04149d80325546f2ae651c03a3597f10.plex.direct
...
08:07:49.287 T:6784    DEBUG <general>: CUtil::ScanForExternalSubtitles: END (total time: 129170 ms)
08:07:49.287 T:6784     INFO <general>: Creating Demuxer
08:07:49.289 T:6784    DEBUG <general>: CDVDDemuxFFmpeg::Open - probing detected format [matroska,webm]
```
I sincerely hope this issue will be fixed with Kodi 19.2 and this pull request: https://github.com/xbmc/xbmc/pull/19974

In the meantime, there's nothing I can do from the PKC side of things. You could try to deactivate scanning for external additional subtitles.

**Comment by yekul** [comment]
Ok great that narrows down the issue at least. Can't find an option to deactivate scanning for external additional subtitles, and have installed no addon that would do that. Any subtitle files have always built contained inside (or for some older stuff next to) existing files. 

As you say this is outside your PKC ballpark now, however i'm wondering if you might know where it's trying to connect to or any ways to bypass this? Wondering if it's something I could set up a firewall rule for so it actually gets a response and moves on quickly rather than just timing out like it seems to.

**Comment by croneter** [comment]
Kodi scans here, where it shouldn't: https://192-168-1-69.04149d80325546f2ae651c03a3597f10.plex.direct:32400/library/parts/21957/1573239338/

Nope sorry, no clue 😞

**Comment by croneter** [comment]
Haha, possible fix yet to be implemented for PKC: https://github.com/plexinc/plex-for-kodi/pull/358

**Comment by croneter** [comment]
Fixed with PKC version 3.4.1
